test/TestServerProjectsLowLevel/head/Generated/HttpSuccessClient.cs

```diff
@@ -159,70 +159,43 @@ public virtual Response Head404(RequestContext context = null)
             request.Uri = uri;
-            message.ResponseClassifier = ResponseClassifier200404.Instance;
             return message;
         }
 
         internal HttpMessage CreateHead204Request(RequestContext context)
         {
-            var message = _pipeline.CreateMessage(context);
+            var message = _pipeline.CreateMessage(context, ResponseClassifier204404);
             var request = message.Request;
             request.Method = RequestMethod.Head;
             var uri = new RawRequestUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/http/success/204", false);
             request.Uri = uri;
-            message.ResponseClassifier = ResponseClassifier204404.Instance;
             return message;
         }
 
         internal HttpMessage CreateHead404Request(RequestContext context)
         {
-            var message = _pipeline.CreateMessage(context);
+            var message = _pipeline.CreateMessage(context, ResponseClassifier204404);
             var request = message.Request;
             request.Method = RequestMethod.Head;
             var uri = new RawRequestUriBuilder();
             uri.Reset(_endpoint);
             uri.AppendPath("/http/success/404", false);
             request.Uri = uri;
-            message.ResponseClassifier = ResponseClassifier204404.Instance;
             return message;
         }
 
-        private sealed class ResponseClassifier200404 : ResponseClassifier
-        {
-            private static ResponseClassifier _instance;
-            public static ResponseClassifier Instance => _instance ??= new ResponseClassifier200404();
-            public override bool IsErrorResponse(HttpMessage message)
-            {
-                return message.Response.Status switch
-                {
-                    200 => false,
-                    404 => false,
-                    _ => true
-                };
-            }
-        }
-        private sealed class ResponseClassifier204404 : ResponseClassifier
-        {
-            private static ResponseClassifier _instance;
-            public static ResponseClassifier Instance => _instance ??= new ResponseClassifier204404();
-            public override bool IsErrorResponse(HttpMessage message)
-            {
-                return message.Response.Status switch
-                {
-                    204 => false,
-                    404 => false,
-                    _ => true
-                };
-            }
-        }
+        private static ResponseClassifier _responseClassifier200404;
+        private static ResponseClassifier ResponseClassifier200404 => _responseClassifier200404 ??= new CoreResponseClassifier(stackalloc int[] { 200, 404 });
+        private static ResponseClassifier _responseClassifier204404;
+        private static ResponseClassifier ResponseClassifier204404 => _responseClassifier204404 ??= new CoreResponseClassifier(stackalloc int[] { 204, 404 });
```

Comment:
@annelo-msft Test case for [Head404Async()](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServerLowLevel.Tests/head.cs#L30) method is failing with these updates. Check [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1404658&view=logs&j=b5cdfb1e-c180-561f-918e-a169e451719e&t=d557719e-b1de-5962-364a-0820efa4a6bb&l=47) in pipeline for the error message.

Reply:
This is issue fixed [here](https://github.com/Azure/azure-sdk-for-net/pull/27334).